ignf_gpf_sdk/workflow/action/ProcessingExecutionAction.py

```diff
@@ -199,75 +200,95 @@ def find_stored_data(self, datastore: Optional[str] = None) -> Optional[StoredDa
-                    self.stored_data.api_delete()
-                # transmission de l'interruption
-            raise KeyboardInterrupt() from e
+
+            except KeyboardInterrupt as e:
+                # on appelle la callback de gestion du ctrl-C
+                if ctrl_c_action is None or ctrl_c_action():
+                    # on arrêter le traitement :
+                    # si le traitement est déjà dans un statut terminé, on ne fait rien => transmission de l'interruption
+                    self.processing_execution.api_update()
+                    s_status = self.processing_execution.get_store_properties()["status"]
+                    if s_status in [ProcessingExecution.STATUS_ABORTED, ProcessingExecution.STATUS_SUCCESS, ProcessingExecution.STATUS_FAILURE]:
+                        # traitement terminé. On fait un dernier affichage :
+                        callback_not_null(self.processing_execution)
+                        Config().om.warning("traitement déjà terminé.")
+                        raise
+
+                    # arrêt du traitement
+                    Config().om.warning("Ctrl+C : traitement en cours d’interruption, veuillez attendre...")
+                    self.processing_execution.api_abort()
+                    # attente que le traitement passe dans un statut terminé
+                    self.processing_execution.api_update()
+                    s_status = self.processing_execution.get_store_properties()["status"]
+                    while s_status not in [ProcessingExecution.STATUS_ABORTED, ProcessingExecution.STATUS_SUCCESS, ProcessingExecution.STATUS_FAILURE]:
+                        # On attend 2s
+                        time.sleep(2)
+                        # On met à jour __processing_execution + valeur status
+                        self.processing_execution.api_update()
+                        s_status = self.processing_execution.get_store_properties()["status"]
+                    # traitement terminé. On fait un dernier affichage :
+                    callback_not_null(self.processing_execution)
+                    # si statut Aborted :
+                    # suppression de l'upload ou de la stored data en sortie
+                    if s_status == ProcessingExecution.STATUS_ABORTED and self.output_new_entity:
+                        if self.upload is not None:
+                            Config().om.warning("Suppression de l'upload en cours de remplissage suite à l’interruption du programme")
+                            self.upload.api_delete()
+                        elif self.stored_data is not None:
+                            Config().om.warning("Suppression de la stored-data en cours de remplissage suite à l'interruption du programme")
+                            self.stored_data.api_delete()
+                    # enfin, transmission de l'interruption
+                    raise KeyboardInterrupt() from e
```

Comment:
Est-ce qu'on crée une nouvelle erreur spécifiquement ou c'est une typo ?

```suggestion
                    raise
```

Reply:
Est-ce qu'on crée une nouvelle erreur spécifiquement ou c'est une typo ?
@LudivineSchlegel ?

Reply:
tu peux raise l'erreur initiale (ctrl+c) donc soit "raise" ou "raise e" c'est la même chose.